segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -38,29 +38,28 @@
 @Slf4j
 public class IndexAppendProcessor {
     static final Duration TIMEOUT = Duration.ofMinutes(1);
-    private final ScheduledExecutorService indexSegmentUpdateExecutor;
     private final StreamSegmentStore store;
+    private final MultiKeyLatestItemSequentialProcessor<String, Long> appendProcessor;
 
     public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor, StreamSegmentStore store) {
-        this.indexSegmentUpdateExecutor = indexSegmentUpdateExecutor;
         this.store = store;
+        appendProcessor = new MultiKeyLatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
     }
 
     /**
      * Appends index segment on a separate thread.
      * @param segmentName  segment name.
+     * @param maxAllowedSize  Maximum allowed event size.
      */
-    protected void processAppend(String segmentName) {
+    protected void processAppend(String segmentName, long maxAllowedSize) {
         // Not updating index segment for transient and transaction type.
         if (isTransientSegment(segmentName) || isTransactionSegment(segmentName)) {
             return;
         }
-        LatestItemSequentialProcessor<String> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
-        latestItemProcessor.updateItem(segmentName);
+        appendProcessor.updateItem(segmentName, maxAllowedSize);
     }
 
-    private void handleIndexAppend(String segmentName) {
-        //TODO : Update the index segment Attribute as needed going forward. If eventcount for main segment is null then we are passing it as 0 in index segment.
+    private void handleIndexAppend(String segmentName, long maxAllowedSize) {
```

Comment:
This logic is incorrect. 
The append needs to be conditional on the count being higher. This means the `attributes` below needs to be setting the value from the `info.getAttributes()`.

Reply:
This logic is handled in PR #7199 .